# Métodos iterativos para determinar valores característicos
Estos métodos son utilizados comúnmente cuando se desea conocer el valor característico de mayor o menor valor absoluto de una matriz. Una ventaja que  presentan estos métodos es que proporcionan simultáneamente el correspondiente vector característico asosiado.

## Método de las potencias
El procedimiento consiste en utilizar la ecuación:

$$ \text{A}\bar{x}=\lambda\bar{x} $$

en forma recursiva, tomando un vector inicial $\bar{x}^{(0)} \neq 0$ de la forma

$$ \bar{x}^{(0)} = \begin{bmatrix}
x_1 \\
x_2 \\
\vdots \\
x_n
\end{bmatrix} $$

Al sustituir este vector en el primer miembro de la ecuación $ \text{A}\bar{x}=\lambda\bar{x} $ y efectuar la multiplicación indicada se obtiene una primera aproximación en el segundo miembro; esto es:

$$ \text{A}\bar{x}^{(0)}=\lambda^{(1)}\bar{x}^{(1)} $$

donde

$$ \lambda^{(1)}\bar{x}^{(1)} = \lambda^{(1)}\begin{bmatrix}
x_1 \\
x_2 \\
\vdots \\
x_n
\end{bmatrix} $$

es el vector del producto realizado.

El siguiente paso consiste en seleccionar el mayor elemento, en valor absoluto, y de este vector y asignarlo como una aproximación de $\lambda^{(1)}$. **La asignación se debe efectuar con su signo correspondiente, aun cuando se haya seleccionado en valor absoluto.**

Posteriormente se *normaliza*  el vector obtenido, lo cual consiste en dividirlo entre $\lambda^{(1)}$; de esta manera se obtiene la primera aproximación al vector característico $\bar{x}^{(1)}$ se tiene que
$$ \text{A}\bar{x}^{(1)}=\lambda^{(2)}\bar{x}^{(2)} $$

Por iteraciones sucesivas se obtiene:
$$ \text{A}\bar{x}^{(2)}=\lambda^{(3)}\bar{x}^{(3)}\\ $$ \text{A}\bar{x}^{(3)}=\lambda^{(4)}\bar{x}^{(4)} \\ \vdots$$

En general
$$ \text{A}\bar{x}^{(k-1)}=\lambda^{(k)}\bar{x}^{(k)} \\
k=1,2,3,4,...$$

Este proceso se repetirá hasta que la diferencia, en valor absoluto, entre los valores característicos obtenidos en dos interaciones sucesivas sea menor que una toleracia preestablecida.

Cabe aclarar que el vector inicial $\bar{x}^{(0)}$ puede ser cualquiera diferente de cero; por ejemplo
$$ \bar{x}^{(0)} = \begin{bmatrix}
1 \\
0 \\
\vdots \\
0
\end{bmatrix} $$
o bien
$$ \bar{x}^{(0)} = \begin{bmatrix}
1 \\
1 \\
\vdots \\
1
\end{bmatrix} $$

### Ejemplo
Dado el sistema:
$$
    \left \{
        \begin{array}{rrcr}
            2x_1  & +4x_2 & = & 0 \\
             x_1  & +3x_2 & = & 0 \\
        \end{array} 
    \right .
$$
obtener de la matriz de coeficientes, el mayor valor característico y su correspondiente vector por el método de las potencias, utilizando como vector inicial 
$$ \bar{x}^{(0)} = \begin{bmatrix}
1 \\
1 \\
\end{bmatrix} $$

#### Solución
El sistema expresado en la forma de la ecuación $ \text{A}\bar{x}=\lambda\bar{x} $ queda:

$$\begin{bmatrix}
2 & 4 \\
1 & 3 \\
\end{bmatrix}
\begin{bmatrix}
x_1 \\
x_2 \\
\end{bmatrix}
=
\lambda \begin{bmatrix}
x_1 \\
x_2 \\
\end{bmatrix}$$

Se utiliza la ecuación $\text{A}\bar{x}^{(k-1)}=\lambda^{(k)}\bar{x}^{(k)}$ para k=1:

$$ \text{A}\bar{x}^{(0)}=\lambda^{(1)}\bar{x}^{(1)}$$

In [2]:
A = [2 4 ; 1 3]

A =

   2   4
   1   3



In [3]:
x0 = [ 1 ; 1]

x0 =

   1
   1



In [4]:
x1 = A * x0

x1 =

   6
   4



Se sustituyen valores:

$$\begin{bmatrix}
2 & 4 \\
1 & 3 \\
\end{bmatrix}
\begin{bmatrix}
1 \\
1 \\
\end{bmatrix}
=
\begin{bmatrix}
6 \\
4 \\
\end{bmatrix}$$

Para normalizar se factoriza en el segundo miembro:

In [5]:
x1 = x1 ./ 6

x1 =

   1.0000
   0.6667



$$\begin{bmatrix}
2 & 4 \\
1 & 3 \\
\end{bmatrix}
\begin{bmatrix}
1 \\
1 \\
\end{bmatrix}
= 6
\begin{bmatrix}
1.0000 \\
0.6667 \\
\end{bmatrix}$$

Por lo tanto:
$$\lambda^{(1)}=6 ; \bar{x}=\begin{bmatrix}
1.0000 \\
0.6667 \end{bmatrix}$$

Para $k=2$ se utiliza la misma ecuación:

In [6]:
x2 = A * x1

x2 =

   4.6667
   3.0000



$$ \begin{bmatrix}
2 & 4 \\
1 & 3 \\
\end{bmatrix}
\begin{bmatrix}
1.0000 \\
0.6667 \\
\end{bmatrix}
=
\begin{bmatrix}
4.6667 \\
3.0000 \\
\end{bmatrix}
$$
Se normaliza nuevamente:

In [7]:
x2 = x2 ./ 4.6667

x2 =

   1.0000
   0.6429



$$ \begin{bmatrix}
2 & 4 \\
1 & 3 \\
\end{bmatrix}
\begin{bmatrix}
1.0000 \\
0.6667 \\
\end{bmatrix}
= 4.6667
\begin{bmatrix}
1.0000 \\
0.6429 \\
\end{bmatrix}
$$
por lo tanto 
$$ \lambda^{(2)}=4.6667;\bar{x}^{(2)} = \begin{bmatrix}
1.0000 \\
0.6429 \\
\end{bmatrix}
$$

para $k=3$ :

In [9]:
x3 = A * x2

x3 =

   4.5714
   2.9286



In [10]:
x3 = x3 ./ 4.5714

x3 =

   1.0000
   0.6406



para $k=4$ :

In [11]:
x4 = A * x3

x4 =

   4.5625
   2.9219



In [12]:
x4 = x4 ./ 4.5625

x4 =

   1.0000
   0.6404



para $k=5$ :

In [13]:
x5 = A * x4

x5 =

   4.5616
   2.9212



In [14]:
x5 = x5 ./ 4.5616

x5 =

   1.0000
   0.6404



para $k=6$ :

In [15]:
x6 = A * x5

x6 =

   4.5616
   2.9212



In [16]:
x6 = x6 ./ 4.5616

x6 =

   1.0000
   0.6404



$$ \lambda^{(6)}=4.5616;\bar{x}^{(6)} = \begin{bmatrix}
1.0000 \\
0.6404 \\
\end{bmatrix}
$$

Por lo tanto, el mayor valor característico con cuatro cifras decimales exactas es:
$$ \lambda=4.5616 $$
y su vector asociado:
$$ \bar{x} = \begin{bmatrix}
1.0000 \\
0.6404 \\
\end{bmatrix}
$$

## Menor valor característico 
Para obtener el menor valor característico se premultiplica la ecuación $\text{A}\bar{x}=\lambda\bar{x}$ por la inversa de A:
$$\text{A}^{-1}\text{A}\bar{x}=\text{A}^{-1}\lambda\bar{x} \\
\bar{x}=\lambda\text{A}^{-1}\bar{x}$$
se divide entre $\lambda$ y se expresa en forma recursiva:
$$ \text{A}^{-1}\bar{x}^{(k-1)}=(\frac{1}{\lambda^{(k)}})\bar{x}^{(k)} \\
k=1,2,3,4,...$$
Esta ecuación es similar a la que se obtuvo para determinar el mayor valor característico, utilizando en esta ocasión la inversa de la matriz de coeficientes y el recíproco de $\lambda$

### continuando con el ejemplo
Obtener el menor valor característico y su vector correspondiente.

In [35]:
A

A =

   2   4
   1   3



Se obtiene su inversa

In [36]:
A1 = inv(A)

A1 =

   1.5000  -2.0000
  -0.5000   1.0000



In [37]:
x = [ 1 ; 1]

x =

   1
   1



Se utiliza la ecuacion $\text{A}^{-1}\bar{x}^{(k-1)}=(\frac{1}{\lambda^{(k)}})\bar{x}^{(k)}$ para k = 1

In [38]:
x1 = A1 * x

x1 =

  -0.5000
   0.5000



In [39]:
x1 = x1 ./ (-0.5)

x1 =

   1
  -1



In [41]:
1 / (-0.5)

ans = -2


In [42]:
x2 = A1 * x1

x2 =

   3.5000
  -1.5000



In [43]:
x2 = x2 ./ 3.5

x2 =

   1.0000
  -0.4286



In [44]:
1 / (3.5)

ans = 0.2857


In [45]:
x3 = A1 * x2

x3 =

   2.3571
  -0.9286



In [46]:
x3 = x3 ./ 2.3571

x3 =

   1.0000
  -0.3939



In [47]:
x4 = A1 * x3

x4 =

   2.2879
  -0.8940



In [48]:
x4 = x4 ./ 2.2879

x4 =

   1.0000
  -0.3907



In [49]:
x5 = A1 * x4

x5 =

   2.2815
  -0.8907



In [50]:
x5 = x5 ./ 2.2815

x5 =

   1.0000
  -0.3904



In [51]:
x6 = A1 * x5

x6 =

   2.2808
  -0.8904



In [52]:
x6 = x6 ./ 2.2808

x6 =

   1.0000
  -0.3904



In [53]:
x7 = A1 * x6

x7 =

   2.2808
  -0.8904



In [54]:
x7 = x7 ./ 2.2808

x7 =

   1.0000
  -0.3904



In [55]:
1/2.2808

ans = 0.4384


Por lo tanto, el menor valor característico, con cuatro cifras decimales exactas, es:
$$\lambda=\frac{1}{2.2808}=0.4384$$
y su vector asociado:
$$\bar{x}=\begin{bmatrix}
1.0000 \\
-0.3904 \end{bmatrix}$$